fhttp/http.go

```diff
@@ -1268,17 +1268,24 @@ func DebugHandler(w http.ResponseWriter, r *http.Request) {
 }
 
 // Serve starts a debug / echo http server on the given port.
-// TODO: make it work for port 0 and return the port found and also
-// add a non blocking mode that makes sure the socket exists before returning
-func Serve(port, debugPath string) {
+// Returns the port that opened for listing socket
+func Serve(port, debugPath string) string {
```

Comment:
let's return the port as a number (why not?)

also maybe a follow up is to refactor common parts with  DynamicHTTPServer() above
https://github.com/istio/fortio/blob/master/fhttp/http.go#L1132

and add a test/exercise the return value